cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_nodegroup.go

```diff
@@ -111,17 +126,21 @@ func (ng *nodegroup) DeleteNodes(nodes []*apiv1.Node) error {
 		}
 
 		machine.Annotations[machineDeleteAnnotationKey] = time.Now().String()
-
 		if _, err := ng.machineapiClient.Machines(machine.Namespace).Update(machine); err != nil {
-			return fmt.Errorf("failed to update machine %s/%s: %v", machine.Namespace, machine.Name, err)
+			return err
+		}
+
+		if err := ng.scalableResource.SetSize(int32(replicas - 1)); err != nil {
+			delete(machine.Annotations, machineDeleteAnnotationKey)
+			// Ignore any errors from Update()
+			ng.machineapiClient.Machines(machine.Namespace).Update(machine)
```

Comment:
Trying to unsuccessfully undo the annotation does not worsen the logic. So it's fine here.

It might be also useful to log the error for debugging purposes.

Reply:
Now logs a warning.